### PR TITLE
KOGITO-633: Management console app routes protected by keycloak authe…

### PR DESCRIPTION
**Pull request by nmirasch**
…ntication

### CODE REVIEW

**Comment by nmirasch** [comment]
@cristianonicolai  @Sara4994 This is the first approach of keycloak auth integration. 
This code forces the login on keycloak before showing anything. The logout and the user name retrieval is added in this commit too. To maintain the keycloak instance across the app, the react-redux useSelector hook. Please could you take a look? any thought is very welcome, Thanks!

**Comment by nmirasch** [comment]
@Sara4994 removed redux, relying on localstorage and recreating keycloak instance on logout.

**Review by AjayJagan — DISMISSED** [review]

**Comment by nmirasch** [comment]
Updated. By default auth is disabled.The keycloak auth configuration have been added at webpack.commons.js using EnvironmentPlugin.

@AjayJagan sorry but I missed your review :(  I have noticed that just once this changes addition and can't find anything there.
@AjayJagan @cristianonicolai @Sara4994 Could you mind take a look on this? Thanks!

**Review by AjayJagan — DISMISSED** [review]

**Comment by nmirasch** [comment]
@cristianonicolai @AjayJagan @Sara4994  Added the configuration of dataindex_url and the token addition when auth is enabled.. I put two commits to be able to try it together. Could you give this a try?
Just uncomment at webpack.common.js AUTH_ENABLED and DATAINDEX_URL to try.

**Comment by nmirasch** [comment]
@cristianonicolai @AjayJagan  @Sara4994 Simplified to one commit. Renamed environment variables. To try it against "localhost:8180/graphql" this fix https://github.com/kiegroup/kogito-apps/pull/42 have to be included

**Comment by nmirasch** [comment]
rebased!

File: packages/management-console/webpack.common.js [hunk 1]
```diff
@@ -12,6 +13,13 @@ module.exports = {
     new HtmlWebpackPlugin({
       template: path.resolve(__dirname, 'src', 'index.html'),
       favicon: 'src/favicon.ico'
+    }),
+    new webpack.EnvironmentPlugin({
+        //OIDC_ENABLED: true,
+        KEYCLOAK_REALM: "kogito",
```
Comment (cristianonicolai) [inline packages/management-console/webpack.common.js]:
```suggestion
        KOGITO_KEYCLOAK_REALM: "kogito",
```

File: packages/management-console/webpack.common.js [hunk 2]
```diff
@@ -12,6 +13,13 @@ module.exports = {
...
+        KEYCLOAK_URL: "http://localhost:8280/auth",
```
Comment (cristianonicolai) [inline packages/management-console/webpack.common.js]:
```suggestion
        KOGITO_KEYCLOAK_URL: "http://localhost:8280",
```

File: packages/management-console/webpack.common.js [hunk 3]
```diff
@@ -12,6 +13,13 @@ module.exports = {
...
+        KEYCLOAK_CLIENT_ID: "kogito-management-console",
```
Comment (cristianonicolai) [inline packages/management-console/webpack.common.js]:
```suggestion
        KOGITO_KEYCLOAK_CLIENT_ID: "kogito-management-console",
```

File: packages/management-console/webpack.common.js [hunk 4]
```diff
@@ -12,6 +13,13 @@ module.exports = {
...
+        //KOGITO_DATAINDEX_URL: "http://localhost:8180/graphql"
```
Comment (cristianonicolai) [inline packages/management-console/webpack.common.js]:
```suggestion
        //KOGITO_DATAINDEX_HTTP_URL: "http://localhost:8180"
```

File: packages/management-console/webpack.common.js [hunk 5]
```diff
@@ -12,6 +13,13 @@ module.exports = {
     new HtmlWebpackPlugin({
       template: path.resolve(__dirname, 'src', 'index.html'),
       favicon: 'src/favicon.ico'
+    }),
+    new webpack.EnvironmentPlugin({
+        //OIDC_ENABLED: true,
```
Comment (cristianonicolai) [inline packages/management-console/webpack.common.js]:
```suggestion
        //KOGITO_OIDC_ENABLED: true,
```

Comment (cristianonicolai) [reply packages/management-console/webpack.common.js]: OIDC or AUTH?

File: packages/management-console/webpack.common.js (hunk 4 above)
Comment (cristianonicolai) [reply packages/management-console/webpack.common.js]: @nmirasch I explicitly removed the context /auth and /graphql from the environment variables. We need to add it manually as the Operator won't set the context, only the service URL.

**Comment by cristianonicolai** [comment]
@nmirasch is it possible to run without comment/uncomment perhaps adding a new run configuration to package.json where these parameters are set?

File: packages/management-console/webpack.common.js (hunk 2 above)
Comment (nmirasch) [reply packages/management-console/webpack.common.js]: done

File: packages/management-console/webpack.common.js (hunk 5 above)
Comment (nmirasch) [reply packages/management-console/webpack.common.js]: I prefer KOGITO_AUTH_ENABLED, I just was trying to be aligned with the one used at dataindex service quarkus.oidc.enabled. What do you think is the best option?

**Comment by nmirasch** [comment]
@cristianonicolai @AjayJagan @Sara4994  I have added different script to be able to enable auth and define different dataindex url.
To just enable keycloak auth on dev: yarn run dev-auth 
The other two scripts: dev-remote-dataindex and dev-remote-dataindex-auth allow to try with a remote data index

File: packages/management-console/webpack.common.js (hunk 5 above)
Comment (cristianonicolai) [reply packages/management-console/webpack.common.js]: keeping aligned would be nice, but I finding weird to use oidc here... we could support other ways of authentication. Would suggest to use auth for now.

File: README.md [hunk 6]
```diff
@@ -29,4 +29,36 @@ yarn run build # skips integration tests and production packing
...
\ No newline at end of file
+> Final artifacts will be on `packages/*/dist` directories.
+
+
+### Starting and Configuring the Keycloak Server
+
+To start a Keycloak Server you can use Docker and just run the following command:
+
+```
+docker run --name keycloak -e KEYCLOAK_USER=admin -e KEYCLOAK_PASSWORD=admin -p 8280:8080 quay.io/keycloak/keycloak
```
Comment (cristianonicolai) [inline README.md]: -e KEYCLOAK_IMPORT also helps to setup the realm without manual steps :)

**Comment by cristianonicolai** [comment]
@nmirasch looks good :)

**Review by cristianonicolai — APPROVED** [review]

**Comment by nmirasch** [comment]
@cristianonicolai  rebased and updated the readme to add the keycloak_import parameter avoiding manual steps importing kogito-realm.

**Comment by nmirasch** [comment]
jenkins retest this please

**Comment by nmirasch** [comment]
@AjayJagan @Sara4994 @cristianonicolai  it's green! :)

**Review by Sara4994 — APPROVED** [review]

**Pull request merged**